src/typechecker.erl

```diff
@@ -748,6 +748,53 @@ negate_bool_type({atom, P, false}) ->
 negate_bool_type(Ty) ->
     Ty.
 
+-spec is_list_type(type()) -> boolean().
+is_list_type({type, _, Name, _}) ->
```

Comment:
This one could perhaps be moved to the `typelib` module (along with `is_int_type/1`). Can be done later.